packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/ClusterManagersController.m

```diff
@@ -0,0 +1,157 @@
+  }
+}
+
+- (void)getClustersWithIdentifier:(NSString *)identifier result:(FlutterResult)result {
+  GMUClusterManager *clusterManager = [self.clusterManagerIdToManagers objectForKey:identifier];
+  if (clusterManager) {
+    NSMutableArray *response = [[NSMutableArray alloc] init];
+
+    NSUInteger integralZoom = (NSUInteger)floorf(_mapView.camera.zoom + 0.5f);
+    NSArray<id<GMUCluster>> *clusters = [clusterManager.algorithm clustersAtZoom:integralZoom];
+    for (id<GMUCluster> cluster in clusters) {
+      if ([cluster.items count] == 0) {
+        continue;
+      }
+
+      GMSMarker *firstMarker = (GMSMarker *)cluster.items[0];
+      NSArray *firstMarkerUserData = firstMarker.userData;
+      if ([firstMarkerUserData count] != 2) {
+        continue;
+      }
+
+      NSString *clusterManagerId = firstMarker.userData[1];
+      if (clusterManagerId == (id)[NSNull null]) {
+        continue;
+      }
+
+      NSMutableArray *markerIds = [[NSMutableArray alloc] init];
+      GMSCoordinateBounds *bounds = [[GMSCoordinateBounds alloc] init];
+
+      for (GMSMarker *marker in cluster.items) {
+        NSString *markerId = marker.userData[0];
+        [markerIds addObject:markerId];
+        bounds = [bounds includingCoordinate:marker.position];
+      }
+
+      [response addObject:@{
+        @"clusterManagerId" : clusterManagerId,
+        @"position" : [FLTGoogleMapJSONConversions arrayFromLocation:cluster.position],
+        @"bounds" : [FLTGoogleMapJSONConversions dictionaryFromCoordinateBounds:bounds],
+        @"markerIds" : markerIds
+      }];
+    }
+    result(response);
+  } else {
+    result([FlutterError errorWithCode:@"Invalid clusterManagerId"
+                               message:@"getClusters called with invalid clusterManagerId"
+                               details:nil]);
+  }
+}
+
+- (bool)didTapCluster:(GMUStaticCluster *)cluster {
+  if ([cluster.items count] == 0) {
+    return NO;
+  }
+
+  GMSMarker *firstMarker = (GMSMarker *)cluster.items[0];
+  NSArray *firstMarkerUserData = firstMarker.userData;
+  if ([firstMarkerUserData count] != 2) {
+    return NO;
+  }
+
+  NSString *clusterManagerId = firstMarker.userData[1];
+  if (clusterManagerId == (id)[NSNull null]) {
+    return NO;
+  }
+
+  NSMutableArray *markerIds = [[NSMutableArray alloc] init];
+  GMSCoordinateBounds *bounds = [[GMSCoordinateBounds alloc] init];
+
+  for (GMSMarker *marker in cluster.items) {
+    NSString *markerId = marker.userData[0];
+    [markerIds addObject:markerId];
+    bounds = [bounds includingCoordinate:marker.position];
+  }
+
+  [self.methodChannel
+      invokeMethod:@"cluster#onTap"
+         arguments:@{
+           @"clusterManagerId" : clusterManagerId,
+           @"position" : [FLTGoogleMapJSONConversions arrayFromLocation:cluster.position],
+           @"bounds" : [FLTGoogleMapJSONConversions dictionaryFromCoordinateBounds:bounds],
+           @"markerIds" : markerIds
+         }];
+  return NO;
```

Comment:
Did you mean to return YES here?

Reply:
Decided to remove return type from this completely and return NO in the caller method. If YES is returned then the cluster would not be focused. Added comment about this to caller method.